geest/gui/panels/tree_panel.py

```diff
@@ -905,9 +1001,154 @@ def run_next_worflow_queue(self):
             return
         # pop the first item from the queue
         next_workflow = self.workflow_queue.pop(0)
-        self.start_workflows(type=next_workflow)
+        # Run the next workflow, using skip_completed to control whether completed items are skipped
+        self.start_workflows(type=next_workflow, skip_completed=skip_completed)
         debug_env = int(os.getenv("GEEST_DEBUG", 0))
         if debug_env:
             self.queue_manager.start_processing_in_foreground()
         else:
             self.queue_manager.start_processing()
+
+    def get_workflow_queue(self, scenario):
+        """
+        Get a list of workflows based on the selected scenario.
+
+        Args:
+            scenario (str): The scenario to execute.
+
+        Returns:
+            list: A list of workflows to be processed.
+        """
+        workflow_queue = []
+
+        # Retrieve all items in the tree model
+        root_item = self.model.rootItem
+        all_items = self.collect_tree_items(root_item)
+
+        # Based on the scenario, filter items
+        if scenario == "Run All":
+            workflow_queue = ["indicators", "factors", "dimensions", "analysis"]
+            QgsMessageLog.logMessage(
+                f"Run All: {workflow_queue}", tag="Geest", level=Qgis.Info
+            )
+
+        elif scenario == "Run All Incomplete":
+            incomplete_items = {
+                "indicators": self.collect_incomplete_items(root_item, "indicator"),
+                "factors": self.collect_incomplete_items(root_item, "factor"),
+                "dimensions": self.collect_incomplete_items(root_item, "dimension"),
+                "analysis": self.collect_incomplete_items(root_item, "analysis"),
+            }
+            # Add only non-empty lists to the queue, maintaining the order
+            for role in ["indicators", "factors", "dimensions", "analysis"]:
+                if incomplete_items[role]:
+                    workflow_queue.append(role)
+            QgsMessageLog.logMessage(
+                f"Run All Incomplete Queue: {workflow_queue}",
+                tag="Geest",
+                level=Qgis.Info,
+            )
+
+        elif scenario == "Run Selected":
+            if self.selected_index is not None:
+                workflow_queue = [all_items[self.selected_index]]
+
+        elif scenario == "Run Selected and All Below It":
+            if self.selected_index is not None:
+                workflow_queue = all_items[self.selected_index :]
+
+        elif scenario == "Run Selected and All Incomplete Below It":
+            if self.selected_index is not None:
+                workflow_queue = [
+                    item
+                    for item in all_items[self.selected_index :]
+                    if item.getStatus() != "✔️"
+                ]
+
+        return workflow_queue
+
+    def collect_tree_items(self, root_item):
```

Comment:
I'm not sure about this here....we have all the items in the tree model, seems like we are just duplicating that?